src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VariantsToTable.java

```diff
@@ -212,9 +209,39 @@ public void onTraversalStart() {
         inputHeader = getHeaderForVariants();
         outputStream = createPrintStream();
 
-        if (genotypeFieldsToTake.isEmpty() && asGenotypeFieldsToTake.isEmpty()) {
-            samples = Collections.emptySortedSet();
-        } else {
+        // if no fields specified, default to include all fields listed in header into table
+        if(fieldsToTake.isEmpty() && genotypeFieldsToTake.isEmpty() && asFieldsToTake.isEmpty() && asGenotypeFieldsToTake.isEmpty()){
+            logger.warn("No fields were specified. All fields will be included in output table.");
+
+            // add all mandatory VCF fields (except INFO)
+            for(VCFHeader.HEADER_FIELDS headerField : VCFHeader.HEADER_FIELDS.values()){
+                if(!headerField.name().equals(VCFHeader.HEADER_FIELDS.INFO.name())) {
+                    fieldsToTake.add(headerField.name());
+                }
+            }
+
+            // add all INFO fields present in VCF header
+            for (final VCFInfoHeaderLine infoLine : inputHeader.getInfoHeaderLines()) {
+                fieldsToTake.add(infoLine.getID());
+            }
+
+            // add all FORMAT fields present in VCF header
+            for (final VCFFormatHeaderLine formatLine : inputHeader.getFormatHeaderLines()) {
+                // ensure GT field listed as first FORMAT field
+                if(formatLine.getID().equals("GT")) {
+                    genotypeFieldsToTake.add(0, formatLine.getID());
+                }
+                else {
+                    genotypeFieldsToTake.add(formatLine.getID());
+                }
+            }
+        }
+
+        // if fields specified, but none are genotype fields, set samples to empty
+        if (genotypeFieldsToTake.isEmpty() && asGenotypeFieldsToTake.isEmpty() && (!fieldsToTake.isEmpty() || !asFieldsToTake.isEmpty())) {
```

Comment:
The `&& (!fieldsToTake.isEmpty() || !asFieldsToTake.isEmpty())` part of this condition here is unnecessary, I believe. If the genotype fields are empty, then either the user originally specified some non-genotype fields, or we hit your new code that populates the non-genotype fields with (at a minimum) the mandatory fields from the VCF.